homeassistant/components/integration/sensor.py

```diff
@@ -174,23 +174,23 @@ def _unit(self, source_unit: str) -> str:
     async def async_added_to_hass(self) -> None:
         """Handle entity which will be added."""
         await super().async_added_to_hass()
-        if state := await self.async_get_last_state():
-            try:
-                self._state = Decimal(state.state)
-            except (DecimalException, ValueError) as err:
-                _LOGGER.warning(
-                    "%s could not restore last state %s: %s",
-                    self.entity_id,
-                    state.state,
-                    err,
-                )
+        if (state := await self.async_get_last_state()) is not None:
+            if state.state == STATE_UNAVAILABLE:
+                self._attr_available = False
             else:
```

Comment:
I think we need to add this, otherwise, we will log an "cannot restore the last state" for "unknown", while we actually can. There is no need to convert this known case into a Decimal.

```suggestion
            elif state.state != STATE_UNKNOWN:
```